src/VisualStudio/Core/Def/Options/PackageSettingsPersister.cs

```diff
@@ -3,54 +3,27 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.SolutionCrawler;
 using Microsoft.VisualStudio.LanguageServices.Setup;
-using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.Threading;
-using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Options
 {
-    internal sealed class PackageSettingsPersister : IOptionPersister
+    internal sealed class PackageSettingsPersister(IGlobalOptionService optionService) : IOptionPersister
     {
-        private readonly IThreadingContext _threadingContext;
-        private readonly IAsyncServiceProvider _serviceProvider;
-        private readonly IGlobalOptionService _optionService;
         private RoslynPackage? _lazyRoslynPackage;
 
-        public PackageSettingsPersister(
-            IThreadingContext threadingContext,
-            IAsyncServiceProvider serviceProvider,
-            IGlobalOptionService optionService)
+        public void Initialize(RoslynPackage package)
         {
-            _threadingContext = threadingContext;
-            _serviceProvider = serviceProvider;
-            _optionService = optionService;
-
-            // Start the process of loading the Roslyn package and getting options, but don't wait for it to complete.
-            // The setting will be refreshed once available.
-            InitializeAsync(_threadingContext.DisposalToken).ReportNonFatalErrorAsync().Forget();
-        }
-
-        private async Task InitializeAsync(CancellationToken cancellationToken)
-        {
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-
-            _lazyRoslynPackage = await RoslynPackage.GetOrLoadAsync(_threadingContext, _serviceProvider, cancellationToken).ConfigureAwait(true);
-            Assumes.Present(_lazyRoslynPackage);
-
-            _optionService.RefreshOption(new OptionKey2(SolutionCrawlerOptionsStorage.SolutionBackgroundAnalysisScopeOption), _lazyRoslynPackage.AnalysisScope);
+            _lazyRoslynPackage = package;
```

Comment:
nit: i would like if we added a check that this happens *once* and dies otherwise.  seems like it would be bad if we got double initialized.